### ISSUE TITLE
Support retroarch

### ISSUE THREAD

**Issue by Schroedingers-Cat** [JosefNemec/Playnite#119]
Support for Retroarch would be great: http://www.retroarch.com/

**Comment by IsamuDragon** [comment]
I would like to second this, since a lot of people use it as an all-in-one emulator

**Comment by JosefNemec** [comment]
I actually thought about it, but I didn't know what to do about it since it's a front end in it self and you can't just pass the ROM file to it via cmdline. You can currently create RetroArch profile manually and provide cmdline arguments as described [here](https://buildbot.libretro.com/docs/guides/cli-intro/). And then treat RetroArch config files as ROMs, it should work then.

It just didn't feel consistent with how other emulators are currently treated in Playnite, so I didn't include RetroArch profile by default. I will have to take a another look at it and somehow integrate it.

**Comment by Schroedingers-Cat** [comment]
According to [this](https://forums.libretro.com/t/running-retroarch-from-command-line/5535/6) you can just pass the rom file via commandl ine.  
Retroarchs config profiles are per core basis and per game basis. Both are saved and named consistently, meaning if you know the name of the core, the path to retroarch and the name of the rom file, you have all the info you need.

**Comment by By6r9** [comment]
Another interesting example: Vice. Emulates bunch of Commodore computers, each with different executable and C64 can be emulated with x64.exe and x64sc.exe.
```
- Name: Vice
  Website: 'http://vice-emu.sourceforge.net/'
  Platforms: [Commodore 64]
    - Name: Vice (fast)
      DefaultArguments: -fullscreen -autostart "{ImagePath}"'
      ImageExtensions: [.bz2, .gz, .d6z, .d7z, .d8z, .g6z, .g7z, .g8z, .d64, .d71, .d80, .d81, .d82, .g64, .g41, .x64, .t64, .p00, .tap, .prg, .zip ]
      ExecutableLookup: ^x64\.exe$
    - Name: Vice (accurate) 
      DefaultArguments: -fullscreen -autostart "{ImagePath}"'
      ImageExtensions: [.bz2, .gz, .d6z, .d7z, .d8z, .g6z, .g7z, .g8z, .d64, .d71, .d80, .d81, .d82, .g64, .g41, .x64, .t64, .p00, .tap, .prg, .zip ]
      ExecutableLookup: ^x64sc\.exe$
  Platforms: [Commodore Plus/4]
    DefaultArguments: -fullscreen -autostart "{ImagePath}"'
    ImageExtensions: [.bz2, .gz, .d6z, .d7z, .d8z, .g6z, .g7z, .g8z, .d64, .d71, .d80, .d81, .d82, .g64, .g41, .x64, .t64, .p00, .tap, .prg, .zip ]
    ExecutableLookup: ^xplus4\.exe$
  
```

**Comment by JosefNemec** [comment]
My proposition:
```
- Name: Vice
  Platforms: [Commodore 64, Commodore Plus/4]
  Website: 'http://vice-emu.sourceforge.net/'
  Configurations:
    - Name: Fast
      DefaultArguments: '-fullscreen -autostart "{ImagePath}"'
      ExecutableLookup: ^x64\.exe$
      ImageExtensions: [.bin, .md]  
    - Name: Accurate
      DefaultArguments: '-fullscreen -autostart "{ImagePath}"'
      ExecutableLookup: ^x64sc\.exe$
      ImageExtensions: [.bin, .md]  
```

In case of RetroArch every configuration would be setting for a different core and when assigning emulator to a game (or doing game import), you would select appropriate core (configuration).

I would keep platforms global for the whole emulator since that IMHO makes sense, because we are defining what all platforms is emulator supporting. Also it would not break current linking where emulators are only provided in "Action" settings based on game's set platform.

**Comment by By6r9** [comment]
You missed in your example xplus4.exe - without linking platform and executable other solution would be  
`name: Vice (Plus/4)
`, and adding another configuration but that is less elegant.

Another can of worms is MAME.
For arcade games it is:
`mame <romname> (or rompath)
`
but executable includes also emulation of approximately thousand of other systems, with different default arguments:

Sony Playstation:
```
mame psu -cdrom <isoname> 
mame psj -cdrom <isoname>
mame pse -cdrom <isoname>
mame psa -cdrom <isoname>
```
for USA, Japan, Europe, Asia variants of bios/system.

Nintendo Game Boy Advance:
`mame gba -cart <romname>
`
etc, so there should be a way to link platform and exe/arguments/extensions.

**Comment by JosefNemec** [comment]
OK I'll move platform definitions to each config section as well. Hopefully that will cover all possible scenarios.

**Comment by JosefNemec** [comment]
Released in 3.0
